source/assets/js/components/fef-swipeable-area.js

```diff
@@ -140,78 +163,117 @@ export class FefSwipeableArea {
-        let visibleAreaLeftEdge = this.$innerContainer.scrollLeft() + INNER_CONTAINER_SCROLL_PADDING,
-            nextItem = this.itemRightPositions.findIndex(rightEdge => rightEdge > visibleAreaLeftEdge),
-            newPosition = this.itemLeftPositions[nextItem] - this.$innerContainer.innerWidth() + INNER_CONTAINER_SCROLL_PADDING;
+        let visibleAreaLeftEdge = this.$innerContainer.scrollLeft(),
+            nextItemIndex = this.itemPositions.findIndex(pos => pos.right > visibleAreaLeftEdge);
+
+        nextItemIndex = Math.max(nextItemIndex - 1, 0);
+
+        let newPosition = this.itemPositions[nextItemIndex].center - (this.$innerContainer.innerWidth() / 2);
 
         this.scrollToPosition(newPosition);
     }
 
     /**
      * Scrolls to a specified position in a specified (or default) time.
-     * Also resets the hinting, if applicable by removing the translation.
+     * Also resets the hinting, if applicable, by removing the translation.
      *
      * @param {Number} position Where to scroll to
      * @param {Number} [time] How long it should take, optional
      */
     scrollToPosition(position, time) {
         time = typeof time === 'undefined' ? DEFAULT_SCROLL_TIME : time;
 
-        this.$innerContainer.children().first().css('transform', 'translateX(0)');
+        this.checkFuturePosition(position);
 
         this.$innerContainer
             .stop(true, false)
-            .animate( { scrollLeft: position }, time, 'easeInOutSine');
+            .animate( { scrollLeft: position }, time, 'easeInOutSine', () => {
+                if (this.isInWideBreakpoints()) {
+                    this.setButtonWidths();
+                }
+            });
+    }
+
+    checkFuturePosition(position) {
+        // If the scroll position *will* be so that it's not possible to
+        // scroll to one direction anymore, remove the hinting. We could
+        // do this in the callback of animate, but if it happens
+        // when starting the animation, it's less janky.
+        if (this.isInWideBreakpoints()) {
+            let willBeOutOfBoundsOnAnySide = false;
+
+            if (position <= 0) {
+                willBeOutOfBoundsOnAnySide = true;
+                this.$buttonBack.removeClass(BUTTON_ACTIVE_CLASS);
+            }
+
+            if (position + this.$innerContainer.innerWidth() >= this.$innerContainer[0].scrollWidth) {
+                willBeOutOfBoundsOnAnySide = true;
+                this.$buttonForward.removeClass(BUTTON_ACTIVE_CLASS);
+            }
+
+            if (willBeOutOfBoundsOnAnySide) {
+                this.applyHint(0);
+            }
+        }
+    }
+
+    setButtonWidths() {
+        let visibleAreaRightEdge = this.$innerContainer.scrollLeft() + this.$innerContainer.innerWidth(),
+            nextItemPos = this.itemPositions.find(pos => pos.right > visibleAreaRightEdge);
+
+        if (nextItemPos) {
+            // add 2 pixels for rounding errors and to hide the shadow of the elements
```

Comment:
Die 2 Pixel könnte man evtl. auch direkt in setButtonWidth() hinzufügen, da Du diese ja immer zur Width addierst.